stan/math/prim/fun/offset_multiplier_constrain.hpp

```diff
@@ -77,21 +75,173 @@ inline return_type_t<T, M, S> offset_multiplier_constrain(const T& x,
  * @throw std::domain_error if mu is not finite
  */
 template <typename T, typename M, typename S>
-inline return_type_t<T, M, S> offset_multiplier_constrain(const T& x,
-                                                          const M& mu,
-                                                          const S& sigma,
-                                                          T& lp) {
-  using std::log;
-  check_finite("offset_multiplier_constrain", "offset", mu);
-  if (sigma == 1) {
-    if (mu == 0) {
-      return identity_constrain(x);
-    }
-    return mu + x;
+inline auto offset_multiplier_constrain(const T& x, const M& mu, const S& sigma,
+                                        return_type_t<T, M, S>& lp) {
+  auto&& mu_ref = to_ref(mu);
+  auto&& sigma_ref = to_ref(sigma);
+  check_finite("offset_multiplier_constrain", "offset", value_of_rec(mu_ref));
+  check_positive_finite("offset_multiplier_constrain", "multiplier",
+                        value_of_rec(sigma_ref));
+  lp += sum(multiply_log(size(x), sigma_ref));
+  return fma(sigma_ref, x, mu_ref);
```

Comment:
Oooof wow yeah this would have been a lot smarter, didn't think of it. I guess since the fma stuff is here already we can keep it though I very much wish I had thought of this